### PR TITLE
exposes cluster/health resource

### PR DESCRIPTION
**Pull request by p0lyn0mial**
**What this PR does / why we need it**: exposes cluster/health resource

### CODE REVIEW

File: api/pkg/api/v1/types.go [hunk 1]
```diff
@@ -269,7 +269,16 @@ type NewClusterStatus struct {
 	URL string `json:"url"`
 }
 
+// NewClusterHealth stores health information about the cluster's components.
+// swagger:model ClusterHealth
+type NewClusterHealth struct {
```
Comment (p0lyn0mial) [inline api/pkg/api/v1/types.go:274]: this is a "virtual" resource, it is not stored in the database thus there is no metadata like `ID`, `Name`, `CreationTimestamp` fields

File: api/pkg/api/v1/types.go [hunk 2]
```diff
@@ -269,7 +269,16 @@ type NewClusterStatus struct {
...
+	Apiserver         bool `json:"apiserver"`
+	Scheduler         bool `json:"scheduler"`
+	Controller        bool `json:"controller"`
+	MachineController bool `json:"machineController"`
+	Etcd              bool `json:"etcd"`
```
Comment (mrIncompetent) [inline api/pkg/api/v1/types.go:279]: I though the idea was to include way more details?

Just encoding the status of a component with `false` or `true` is not enough - as we already see in the dashboard today.
Or are we planning on doing this in another iteration via a `v2` endpoint?

Comment (p0lyn0mial) [reply api/pkg/api/v1/types.go:279]: once the client switches to the new api it cannot read the health from the `cluster` resource (as it was reading before) instead it has to consume `cluster/health` endpoint. My intention was not to break existing functionality thus decided to return what we have. 

My plan was to change the implementation of the endpoint in a separate PR.
Hopefully the client will not transition to the new API until then, so we wouldn't have to introduce `v2` endpoint.

Comment (p0lyn0mial) [reply api/pkg/api/v1/types.go:279]: If we are in agreement then I'll create issue for that.

Comment (mrIncompetent) [reply api/pkg/api/v1/types.go:279]: Please create an issue for that :)

**Review by mrIncompetent — APPROVED** [review]

File: api/pkg/api/v1/types.go (hunk 2 above)
Comment (p0lyn0mial) [reply api/pkg/api/v1/types.go:279]: https://github.com/kubermatic/kubermatic/issues/1485

could you help defining the new metrics ?

**Comment by p0lyn0mial** [comment]
I'm merging this branch into https://github.com/kubermatic/kubermatic/pull/1471

**Pull request merged**